### PR TITLE
cdc: no more old value cache when all downstream deregistered

### PR DESCRIPTION
**Pull request by hi-rustin**
Signed-off-by: hi-rustin <rustin.liu@gmail.com>

### What is changed and how it works?

Close https://github.com/tikv/tikv/issues/10091

What's Changed:

### Related changes

None

### Check List 

Tests 

- Unit test
- [x] [Manual test](https://github.com/tikv/tikv/pull/11876#issuecomment-1016052750)

Side effects

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
I will do the manual test tomorrow.

**Comment by hi-rustin** [comment]
before: 
image

after: 
image

Test steps:
1. create a changefeed with old value enabled.
2. sysbench --db-driver=mysql --mysql-host=172.16.5.33 --mysql-port=47904 --mysql-user=root --time=15000 --report-interval=10 --threads=64 --mysql-db=test --tables=1 --table-size=6000000 oltp_write_only prepare && \
sleep 60 && sysbench --db-driver=mysql --mysql-host=172.16.5.33 --mysql-port=47904 --mysql-user=root --time=15000 --report-interval=10 --threads=64 --mysql-db=test --tables=1 --table-size=6000000 oltp_write_only run
3. kill -SIGSTOP CDCPID

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -1435,8 +1435,10 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Runnable for Endpoint<T, E> {
                 cb();
             }
             Task::TxnExtra(txn_extra) => {
-                for (k, v) in txn_extra.old_values {
-                    self.old_value_cache.cache.insert(k, v);
+                if !self.capture_regions.is_empty() {
+                    for (k, v) in txn_extra.old_values {
```
Comment (overvenus) [inline components/cdc/src/endpoint.rs:1439]: Can we stop taking old_values in txn layer when there is no downstream registered?

**Pull request closed without merging**